### ISSUE TITLE
Handle Compose stability

### ISSUE THREAD

**Issue by BoD** [apollographql/apollo-kotlin#5732]
The models generated by Apollo Kotlin are immutable, and therefore should participate in skipping recomposition. However list fields are generated with `List<T>` which is considered unstable by Compose.

More information about Compose stability [here](https://developer.android.com/jetpack/compose/performance/stability).

We could mark the generated models with [`@Immutable`](https://developer.android.com/reference/kotlin/androidx/compose/runtime/Immutable) but that implies with a dependency to `androidx.compose.runtime:runtime`, so we would need to make this configurable

In the meantime users can already mark the generated models as stable manually with the [Compose compiler configuration file](https://developer.android.com/jetpack/compose/performance/stability/fix#configuration-file)
- globally if they are generated in a dedicated package (`com.example.models.**`)
- by adding all models individually (albeit this is tedious and error-prone)

**Comment by BoD** [comment]
Thanks a lot for sharing, interesting stuff! From what I gather, it sounds like with this change, stability won't be as important as it was.

It still doesn't hurt and is pretty easy to mark all generated classes as stable with the configuration file. Maybe this ticket is just to document how to do this somewhere in our docs.

**Comment by StylianosGakis** [comment]
Yeah that is fair, as long as you do not mind having the annotation in the sources, and you anyway guarantee that those classes are immutable, then adding the annotation is also safe enough that it shouldn't matter.

But I agree with keeping the ticket open. Strong skipping mode isn't on by default yet anyway.
